### PR TITLE
[rustdoc] Fix display of features

### LINKED ISSUE #118615: `doc_auto_cfg` formatting is not that great
> `cargo doc` has a new `doc_auto_cfg` feature that finally seems to figure this out based on `cfg` attributes so we don't have to generate the redundant `doc` comments any longer. [rust-lang/rust#43781](https://github.com/rust-lang/rust/issues/43781)
> 
> Unfortunately, formatting is not that great.
> 
> image

*[originally posted](https://github.com/microsoft/windows-rs/issues/2717#issuecomment-1838819740) by @kennykerr*

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes https://github.com/rust-lang/rust/issues/118615.

It now looks like this:

image

We can't use flex without breaking the flow, meaning we can't vertically align items as we want. Because of that, the `min-height` was problematic as it rendered weirdly and therefore needed to be removed.

r? @notriddle

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
With `line-height: 36px;`:

image

File: src/librustdoc/html/static/css/rustdoc.css [hunk 1]
```diff
@@ -1100,11 +1094,9 @@ so that we can apply CSS-filters to change the arrow color in themes */
 	font-weight: normal;
 	color: var(--main-color);
 	background-color: var(--stab-background-color);
-	width: fit-content;
```
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/librustdoc/html/static/css/rustdoc.css:1103]: It wasn't in use said firefox, so I removed it.

**Comment by GuillaumeGomez** [comment]
> Right, but the point of me asking is that the ones with the 👎 should be the same height as the ones without any emoji in them, as well as items like this:
> 
> image

I explained my reasoning here:

> We can't use flex without breaking the flow, meaning we can't vertically align items as we want. Because of that, the `min-height` was problematic as it rendered weirdly and therefore needed to be removed.

File: src/librustdoc/html/static/css/rustdoc.css (hunk 1 above)
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/librustdoc/html/static/css/rustdoc.css:1103]: Ah then we know why I changed it. :D

[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/librustdoc/html/static/css/rustdoc.css:1103]: Oh you're right. No I didn't meant to change it. Might be worth doing it though but I'll open another PR for that. Let me revert this change.

**Comment by GuillaumeGomez** [comment]
I reverted the unwanted width change. I think it's worth considering it but let's do that in another PR.

File: src/librustdoc/html/static/css/rustdoc.css (hunk 1 above)
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/librustdoc/html/static/css/rustdoc.css]: No don't worry. Just curious about the why. It looks very bad with it. Even more when there are no emoji.

File: src/librustdoc/html/static/css/rustdoc.css [hunk 2]
```diff
@@ -1081,15 +1081,9 @@ so that we can apply CSS-filters to change the arrow color in themes */
 }
 
 .item-info .stab {
-	/* This min-height is needed to unify the height of the stab elements because some of them
-	   have emojis.
-	*/
-	min-height: 36px;
-	display: flex;
+	display: block;
 	padding: 3px;
```
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/librustdoc/html/static/css/rustdoc.css:1085]: Still looks bad when there is more than one line:

image

**Review by notriddle — CHANGES_REQUESTED** [review]

**Comment by GuillaumeGomez** [comment]
Updated!

**Review by notriddle — APPROVED** [review]

**Pull request merged**